core/types/celo_additions.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+func (h *Header) IstanbulExtra() (*IstanbulExtra, error) {
+	if h.deserializedExtra == nil {
+		extra, err := ExtractIstanbulExtra(h)
```

Comment:
Since `ExtractIstanbulExtra` is now only used in `types` we could make it unexported which would force code to only use this function on the header.